components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -273,18 +286,41 @@ open class GleanInternalAPI internal constructor () {
             return
         }
 
+        if (!uploadEnabled) {
+            logger.error("Glean must be enabled before handling events.")
+            return
+        }
+
+        var availableToSend = false
+
         when (pingEvent) {
             Glean.PingEvent.Background -> {
-                sendPing(BaselinePing.STORE_NAME, BaselinePing.STORE_NAME)
-                sendPing("events", "events")
+                // Schedule the baseline and event pings and save the result to determine whether or not
+                // we need to schedule the upload worker
+                assembleAndSerializePing(BaselinePing.STORE_NAME)
+                assembleAndSerializePing("events")
+
+                // Always true since baseline is always sent
+                availableToSend = true
             }
             Glean.PingEvent.Default -> {
+                // Check the metrics ping schedule to determine whether it's time to schedule a new
+                // metrics ping or not
                 if (metricsPingScheduler.canSendPing()) {
-                    sendPing(MetricsPingScheduler.STORE_NAME, MetricsPingScheduler.STORE_NAME)
+                    availableToSend = assembleAndSerializePing(MetricsPingScheduler.STORE_NAME)
                     metricsPingScheduler.updateSentTimestamp()
```

Comment:
Should we only update this when `availableToSend` is `true`?  (It's possible I just don't understand the logic...)

Reply:
available to send just means that there was data serialized and we need to schedule the uploader.  If there wasn't something to send, it should be false, but I don't know if that changes when we want to check for a ping to send again.  I'm curious what @Dexterp37 has to say, but you make a good point.